source/as-promise.ts

```diff
@@ -122,10 +125,10 @@ export default function asPromise(options: Options) {
 			'redirect',
 			'uploadProgress',
 			'downloadProgress'
-		].forEach(event => emitter.on(event, (...args) => proxy.emit(event, ...args)));
-	}) as CancelableRequest<IncomingMessage>;
+		].forEach(event => emitter.on(event, (...args: unknown[]) => proxy.emit(event, ...args)));
+	}) as CancelableRequest<Response | Buffer | string | any>;
```

Comment:
`Response | Buffer | string | any` should be a defined type.

And `any` => `unknown`.